keras_nlp/tokenizers/sentence_piece_tokenizer.py

```diff
@@ -0,0 +1,175 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import base64
+
+import tensorflow as tf
+import tensorflow_text as tf_text
+
+from keras_nlp.tokenizers import tokenizer
+
+
+class SentencePieceTokenizer(tokenizer.Tokenizer):
+    """A SentencePiece tokenizer layer.
+
+    This layer provides an implementation of SentencePiece tokenization
+    as described in the [SentencePiece paper](https://arxiv.org/abs/1808.06226)
+    and the [SentencePiece package](https://pypi.org/project/sentencepiece/).
+    The tokenization will run entirely within the Tensorflow graph, and can
+    be saved inside a `keras.Model`.
+
+    By default, the layer will output a `tf.RaggedTensor` where the last
+    dimension of the output is ragged after whitespace splitting and sub-word
+    tokenizing. If `sequence_length` is set, the layer will output a dense
+    `tf.Tensor` where all inputs have been padded or truncated to
+    `sequence_length`. The output dtype can be controlled via the `dtype`
+    argument, which should be either an integer or string type.
+
+    Args:
+        model_file: A path to a SentencePiece serialized model file. One
```

Comment:
merge into one argument, rename from model to something else?

Reply:
Trying to think of names here. "serialized" might be useful because this is either a "serialized" might be useful here as this is either a binary file or a byte string...

`serialized_settings`, `serialized_config`, `serialized_proto`, `serialized_spec`, `settings`, `config`, `proto`, `spec`

Do any of these sound good @fchollet ? Here's the proto message if helpful https://github.com/google/sentencepiece/blob/master/src/sentencepiece_model.proto

Reply:
want to know more about this - are you proposing to keep only one argument, and let the code decide how to load the model?

Reply:
Yeah, this was a not to self written during my meeting with Francois. His preference is to merge the arguments into one (which match TextVectorization and WordPiece). The implementation might be a little more tricky, but the experience more consistent.

Then there's the separate question of what to name. We discussed moving away from the word "model", which means something specific in Keras we do not mean here. (that's the question above)

Reply:
Talked today. Let's make a single overloaded arg called `proto`. This is either a serialized proto bytestring or a proto filepath.